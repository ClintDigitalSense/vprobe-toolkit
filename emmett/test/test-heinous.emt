Arith easyAdd = 1 + 2;
Arithprec prec = 1 + 2 * 3;
Arith precedence = 1 == 2 && 4 == 5;
SimpleAsgn myInt0 = 0xdeadbeef; 	# Trivial assignment
HashPlusPlus aggr0[9]++; 	# Trivial assignment
Aggr aggr0[9] <- 0;		# aggregating assignment
Aggr keylessaggr <- 0;		# keyless aggr
Aggr complicatedAggr[9, "str1", "str2"] <- 145 - 23;

int function_def(int a12) 
{
   easyAdd = 12 + a12; # var defined outside of function, func arg.
   definedInFunc = 0xdeadbeef; # var defined inside function
   strVar = "testareenio!";
   return definedInFunc ^ a12;
}

Block {
	/* Do block statements grossly work? */
	blockaggr[9, "blockkey1"]   <- 12;
	blockscalar =123;
}
If1 if(0) a = 12;
If2 if(1){  aggr1[1,3] <- 2; scale =17; } # if with a block

int nakedPred();
NestBlocks {
	if (nakedPred())
		g=12;
	if (nakedPred()) {
		block0aggr[12, "snarf"] <- TSC;
		block0asgn=12;
	}
	{ nakedBlock0 = 13; }
	
}
