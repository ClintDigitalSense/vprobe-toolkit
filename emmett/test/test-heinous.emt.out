(version 0.2)
(definteger a)
(defaggr aggr0 1 0)
(defaggr aggr1 2 0)
(defaggr block0aggr 1 1)
(definteger block0asgn)
(defaggr blockaggr 1 1)
(definteger blockscalar)
(defaggr complicatedAggr 1 2)
(definteger definedInFunc)
(definteger easyAdd)
(definteger g)
(defaggr keylessaggr 0 0)
(definteger myInt0)
(definteger nakedBlock0)
(definteger prec)
(definteger precedence)
(definteger scale)
(defstring strVar)

 ;;;;;
(vprobe Arith ( setint easyAdd   3))
 ;;;;;
(vprobe Arithprec ( setint prec   7))
 ;;;;;
(vprobe Arith ( setint precedence    (&&  (== 1  2 )   (== 4  5 ) )))
 ;;;;;
(vprobe SimpleAsgn ( setint myInt0   3735928559))
 ;;;;;
(vprobe HashPlusPlus ( aggr  aggr0 ( 9) () 1))
 ;;;;;
(vprobe Aggr ( aggr  aggr0 ( 9) () 0))
 ;;;;;
(vprobe Aggr ( aggr  keylessaggr () () 0))
 ;;;;;
(vprobe Aggr ( aggr  complicatedAggr ( 9) ( "str1" "str2") 122))(defun function_def(a12 )( do  ( setint easyAdd    (+ 12  a12 ))( setint definedInFunc   3735928559)( setstr strVar   "testareenio!")) (^ definedInFunc  a12 ))
 ;;;;;
(vprobe Block ( do  ( aggr  blockaggr ( 9) ( "blockkey1") 12)( setint blockscalar   123)))
 ;;;;;
(vprobe If1 ( cond  (0( setint a   12))))
 ;;;;;
(vprobe If2 ( cond  (1( do  ( aggr  aggr1 ( 1 3) () 2)( setint scale   17)))))
 ;;;;;
(vprobe NestBlocks ( do  ( cond  ( (nakedPred)( setint g   12)))( cond  ( (nakedPred)( do  ( aggr  block0aggr ( 12) ( "snarf") TSC)( setint block0asgn   12))))( do  ( setint nakedBlock0   13))))